src/native/libs/System.Globalization.Native/pal_locale.c

```diff
@@ -134,7 +134,11 @@ int32_t FixupLocaleName(UChar* value, int32_t valueLength)
 // is not desirable at all because it doesn't support case insensitive string comparisons.
 const char* DetectDefaultLocaleName()
 {
+#ifndef __APPLE__
     const char* icuLocale = uloc_getDefault();
+#else
+    const char* icuLocale = DetectDefaultAppleLocaleName();
```

Comment:
should we make this fallback to `uloc_getDefault()` when returning empty locale name? I mean writing the code like:

```
const char* icuLocale;

#ifdef __APPLE__
icuLocale = DetectDefaultAppleLocaleName();
if (icuLocale == NULL || icuLocale[0] = 0)
#endif
    icuLocale = uloc_getDefault();
```